services/QuillLMS/app/services/diagnostic_insights/diagnostics_assigned_by_classroom.rb

```diff
@@ -29,15 +29,20 @@ def run = grouped_results
     private def grouped_results = parsed_results.group_by { |slsa| diagnostic_type_lookup.fetch(slsa[:activity_id]) }
 
     private def parsed_results
-      student_learning_sequence_activities
-        .map do |slsa|
-          {
-            activity_id: slsa.activity_id,
-            completed_at: slsa.completed_at,
-            created_at: slsa.created_at,
-            user_id: slsa.student_learning_sequence.user_id
-          }
-        end
+      filtered_results.map do |slsa|
+        {
+          activity_id: slsa.activity_id,
+          completed_at: slsa.completed_at,
+          created_at: slsa.created_at,
+          user_id: slsa.student_learning_sequence.user_id
+        }
+      end
+    end
+
+    private def filtered_results
```

Comment:
I think this might be better as a scope:

```ruby
class StudentLearningSequenceActivity
  scope :assigned, -> {
    joins(:student_learning_sequence, :classroom_unit)
      .where("student_learning_sequences.user_id = ANY (classroom_units.assigned_student_ids)")
  }
end